handlers/revenue-recogniser-job/src/main/scala/com/gu/recogniser/Handler.scala

```
@@ -34,13 +35,13 @@ object Handler extends RequestStreamHandler {
     } yield ()
     val _ = maybeSuccess.toTry.get // throws exception if something failed
     log("finished successfully - sending metric!")
-    putMetric(stage)
+    putMetric(stage).get
```

Doesn't this `get` undermine the change in `putMetric` result type? ie. it will blow up here rather than in line 44

---

I don't think so - you can treat any type as Unit, so it will just ignore the result rather than automatically calling `.get` on it if it happens to be a Try.  This was causing a warning hence why I noticed.

The reason I went ahead with the change, is if it fails silently, the alarm would indeed go off, but it wouldn't be clear why from the logs.  So this change will mean that once the alarm does go off, we will see in the logs that the lambda exploded and logged an exception, so we can realise what's going on.
This did bite us recently when I accidentally blew the metrics quota, and the IT-Test runner was logging a rate limit exceeded error, but apart from that everything seemed to be working.  Which was very confusing.

---

I have proved it here: https://scalafiddle.io/sf/ijmen2i/1
```scala
import scala.util._
// Start writing your ScalaFiddle code here
def hi(): Unit = Failure(new RuntimeException("boom"))

val x = hi()
println("first is: " + x)


def hi2(): Try[Unit] = Failure(new RuntimeException("boom"))

val xx = hi2().get
println("second is: " + xx)
```
```
first is: undefined
ERROR: undefined
Error
    at $c_jl_RuntimeException.eval [as fillInStackTrace__jl_Throwable] (eval at  (https://scalafiddle.io/resultframe?theme=light:46:37), :2523:14)
.......

```